improved automatic search of subtitles for video without metadata

emavgl opened the pull request:
I've touched the lines below 219
`  if item['title'] == "":`
to improve automatic search of subtitles whenever there is a video without correct metadata.
Using a regex the code parses the name of the file and gets: name of the show, episode and season.
Regex matches the most commons subtitles string forms like:
- Gotham S02E03
- Gotham 2x04  
- ecc

luivit commented on service.py:
```diff
@@ -201,14 +215,29 @@ def get_params():
     item['title']  = normalizeString(xbmc.getInfoLabel("VideoPlayer.Title"))      # no original title, get just Title
-    
+    toParse = item['title'].lower().replace('.', " ");
+    res=re.findall('(.*?)s?0?(\d{1,3})x?e?0?(\d{1,3})', urllib.unquote(toParse), re.IGNORECASE)
```
Use **parseSearchString** even here

luivit commented on service.py:
```diff
@@ -201,14 +215,29 @@ def get_params():
+        item['3let_language']=[]
+        for lang in urllib.unquote(params['languages']).decode('utf-8').split(","):
+            item['3let_language'].append(xbmc.convertLanguage(lang,xbmc.ISO_639_2))
+        Search(item)
```
remove the 233 line (otherwise it is called 2 times)
thx

emavgl commented:
Should be fixed in 902288852cb186a15efd80befa9b1e060cb27748
Give it a look.

luivit commented:
yes, now the code it's ok but regex doesn't work with tvshow that have numbers in title.
(es. the 100, the 4400, 100 Questions, 11.22.63, 12 Monkeys, 1600 Penn, ecc)
you should edit into:
`(.*\d*?)s?0?(\d{1,3})x?e?0?(\d{1,3})`

emavgl commented:
I think that your regex doesn't work too: I tried it on [rubular](http://rubular.com/) and the "Match groups" list isn't what we expected. You can try yourself using the text: _The 100 s01e03_  
I've added a space on it between `(.*\d*?)` and the rest of the regex. It should work now.

Fixed here e3b7e83a842af8e5a0ec4758f5b00e1729ff202b

luivit commented:
yes, i forgot the space while i was writing regex here :)
